tooling/hcpctl/cmd/release/options.go

```diff
@@ -0,0 +1,160 @@
+	cmd.Flags().StringVar(&opts.SdpPipelinesCommit, "sdp-pipelines-commit", opts.SdpPipelinesCommit, "SDP Pipelines commit SHA")
+
+	return nil
+}
+
+// RawReleaseStatusOptions holds input values.
+type RawReleaseStatusOptions struct {
+	ReleaseName        string
+	Namespace          string
+	OutputFormat       string
+	KubeConfig         string
+	KubeContext        string
+	AroHcpCommit       string
+	SdpPipelinesCommit string
+}
+
+// validatedReleaseStatusOptions is a private wrapper that enforces a call of Validate() before Complete() can be invoked.
+type validatedReleaseStatusOptions struct {
+	*RawReleaseStatusOptions
+	settings *cli.EnvSettings
+}
+
+type ValidatedReleaseStatusOptions struct {
+	// Embed a private pointer that cannot be instantiated outside of this package.
+	*validatedReleaseStatusOptions
+}
+
+// completedReleaseStatusOptions is a private wrapper that enforces a call of Complete() before execution can be invoked.
+type completedReleaseStatusOptions struct {
+	ReleaseName        string
+	Namespace          string
+	OutputFormat       string
+	HelmClient         *action.Configuration
+	KubeClient         kubernetes.Interface
+	AroHcpCommit       string
+	SdpPipelinesCommit string
+}
+
+type ReleaseStatusOptions struct {
+	// Embed a private pointer that cannot be instantiated outside of this package.
+	*completedReleaseStatusOptions
+}
+
+func (o *RawReleaseStatusOptions) Validate(ctx context.Context) (*ValidatedReleaseStatusOptions, error) {
+	// Validate output format
+	switch o.OutputFormat {
+	case "yaml", "json":
+		// Valid formats
+	default:
+		return nil, fmt.Errorf("unsupported output format: %s (supported: yaml, json)", o.OutputFormat)
+	}
+
+	// Create Helm settings
+	settings := cli.New()
+
+	// Set kubeconfig if provided
+	if o.KubeConfig != "" {
+		if _, err := os.Stat(o.KubeConfig); os.IsNotExist(err) {
+			return nil, fmt.Errorf("kubeconfig file does not exist: %s", o.KubeConfig)
+		}
+		settings.KubeConfig = o.KubeConfig
+	}
+
+	// Set kube-context if provided
+	if o.KubeContext != "" {
+		settings.KubeContext = o.KubeContext
+	}
+
+	return &ValidatedReleaseStatusOptions{
+		validatedReleaseStatusOptions: &validatedReleaseStatusOptions{
+			RawReleaseStatusOptions: o,
+			settings:                settings,
+		},
+	}, nil
+}
+
+func (o *ValidatedReleaseStatusOptions) Complete(ctx context.Context) (*ReleaseStatusOptions, error) {
+	// Create Helm action configuration
+	actionConfig := new(action.Configuration)
+
+	// Initialize with the appropriate namespace
+	// If no namespace specified, we'll use "" which means all namespaces for discovery
+	namespace := o.Namespace
+	if namespace == "" {
+		// For discovery mode, we need to query all namespaces
+		// Helm will handle this appropriately in the list action
+		namespace = ""
+	}
+
+	if err := actionConfig.Init(o.settings.RESTClientGetter(), namespace, os.Getenv("HELM_DRIVER"), func(format string, v ...interface{}) {
```

Comment:
read the env var when initializing the `opts` and then use options here.  Env vars are hard to track through a program